src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/FlowProcessingAction.java

```diff
@@ -98,10 +121,83 @@ protected FlowPath getFlowPath(PathId pathId) {
         return new HashSet<>(flowRepository.findFlowsIdsByEndpointWithMultiTableSupport(switchId, port));
     }
 
+    protected FlowPathSpeakerView makeFlowPathNewView(Flow flow, FlowPath path, PathResources resources)
+            throws ResourceAllocationException {
+        FlowPathSpeakerView.FlowPathSpeakerViewBuilder pathSnapshot = FlowPathSpeakerView.builder(path)
+                .resources(resources);
+        return pathSnapshot.build();
+    }
+
+    protected FlowPathSpeakerView makeFlowPathOldView(
+            Flow flow, PathId pathId, PathResources resources) {
+        FlowPath path = getFlowPath(pathId);  // need to ensure that all relations are loaded
+        FlowPathSpeakerView.FlowPathSpeakerViewBuilder pathView = FlowPathSpeakerView.builder(path)
+                .resources(resources);
+
+        FlowEndpoint ingressEndpoint = FlowSideAdapter.makeIngressAdapter(flow, path).getEndpoint();
+        SwitchProperties switchProperties = getSwitchProperties(ingressEndpoint.getSwitchId());
+        pathView.removeCustomerPortRule(
+                checkIfSharedCustomerPortForwardingFlowCanBeRemoved(flow.getFlowId(), ingressEndpoint));
+        pathView.removeCustomerPortLldpRule(
+                checkIfSharedLldpPortRuleCanBeRemoved(flow.getFlowId(), ingressEndpoint, switchProperties));
+        pathView.removeCustomerPortArpRule(
+                checkIfSharedArpPortRuleCanBeRemoved(flow.getFlowId(), ingressEndpoint, switchProperties));
+        return pathView.build();
```

Comment:
it works wrong. With such checks update and reroute operations can delete needed shared rules. 

**Test to check:**
1. enable multiTable, switchLldp, switchArp properties for all switches
2. create a flow.
3. updated flow bandwidth via V2 update.
4. validate src and dst switches.

**Expected result:**
successful switch validation

**Actual results:**
3 missing rules (LLDP, ARP, inputCustomer)

**Reason:**
In all checkIfShare* methods you are checking that no other flow uses shared rules. But your current flow still need these shared rules. So you can't remove them